src/m365/spo/commands/site/site-admin-list.ts

```diff
@@ -0,0 +1,262 @@
+
+      const urlFilter = formatting.escapeXml(`Url -eq '${siteUrl}'`);
+      const requestBody: string = `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="2" ObjectPathId="1" /><ObjectPath Id="4" ObjectPathId="3" /><Query Id="5" ObjectPathId="3"><Query SelectAllProperties="false"><Properties /></Query><ChildItemQuery SelectAllProperties="false"><Properties><Property Name="SiteId" ScalarProperty="true" /></Properties></ChildItemQuery></Query></Actions><ObjectPaths><Constructor Id="1" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /><Method Id="3" ParentId="1" Name="GetSitePropertiesFromSharePointByFilters"><Parameters><Parameter TypeId="{b92aeee2-c92c-4b67-abcc-024e471bc140}"><Property Name="Filter" Type="String">${urlFilter}</Property><Property Name="IncludeDetail" Type="Boolean">false</Property><Property Name="IncludePersonalSite" Type="Enum">1</Property><Property Name="StartIndex" Type="String">0</Property></Parameter></Parameters></Method></ObjectPaths></Request>`;
+
+      const requestOptions: CliRequestOptions = {
+        url: `${spoAdminUrl}/_vti_bin/client.svc/ProcessQuery`,
+        headers: {
+          'X-RequestDigest': res.FormDigestValue
+        },
+        data: requestBody
+      };
+
+      const response: string = await request.post<string>(requestOptions);
+      const jsonData: ClientSvcResponse = JSON.parse(response);
+      const responseContent: ClientSvcResponseContents = jsonData[0];
+
+      if (responseContent.ErrorInfo) {
+        throw responseContent.ErrorInfo.ErrorMessage;
+      }
+
+      const sites: SPOSitePropertiesEnumerable = jsonData[jsonData.length - 1];
+      const siteId = sites?._Child_Items_?.[0]?.SiteId ?? undefined;
+
+      if (!siteId) {
+        throw 'Failed to obtain site Id from the provided site URL.';
+      }
+
+      const guid = siteId.replace('/Guid(', '').replace(')/', '');
+
+      if (!validation.isValidGuid(guid)) {
+        throw 'Failed to obtain site Id from the provided site URL.';
+      }
+      return guid;
+    }
+    catch (error) {
+      throw error;
+    }
+  }
+
+  private async getPrimaryAdminLoginNameFromAdmin(adminUrl: string, siteId: string): Promise<string> {
+    try {
+      const requestOptions: CliRequestOptions = {
+        url: `${adminUrl}/_api/SPO.Tenant/sites(%27${siteId}%27)?$select=OwnerLoginName`,
+        headers: {
+          accept: 'application/json;odata=nometadata',
+          'content-type': 'application/json;charset=utf-8'
+        }
+      };
+
+      const response: string = await request.get<string>(requestOptions);
+      const responseContent = JSON.parse(response);
+      return responseContent.OwnerLoginName;
+    }
+    catch (err: any) {
+      throw err;
+    }
```

Comment:
same here. if it's just a small local helper method we don't need to wrap it around a `try catch` that does not do anything more than just `throw`